File: check_outputs.py
from optparse import OptionParser
import glob
import os

def getOptions():
    parser = OptionParser()
    parser.add_option("--f", dest = "indir", help = "Directory containing input files",
                      type = "string")
    parser.add_option("--o", dest = "outdir",
                      help = "Directory containing output files", type = "string")
    (options, args) = parser.parse_args()
    return options

def main():
    options = getOptions()

    indir = options.indir
    if not indir.endswith("/"):
        indir = indir + "/"
    outdir = options.outdir
    if not outdir.endswith("/"):
        outdir = outdir + "/" 

    for infile in glob.glob(indir + "*"):
        pattern = os.path.basename(infile).split(".")
        pattern[0] = "outfile"
        pattern[-1] = "out"
        expected_outfile = outdir  + ".".join(pattern)
        
        if not os.path.exists(expected_outfile):
            print expected_outfile

if __name__ == '__main__':
    main()

File: check_outputs.R
main <- function() {

    load_packages()
    opt <- parse_options()
    indir <- opt$indir
    outdir <- opt$outdir

    files <- list.files(indir)
    for (infile in files ){
        string_parts <- str_split(infile, pattern = "\\.")[[1]]
        string_parts[1] <- "outfile"
        string_parts[length(string_parts)] <- "out"
       
        expected_outfile <- paste(outdir, "/", paste(string_parts, 
                                                     collapse="."),sep="")
        if(!file.exists(expected_outfile)) {
            print(expected_outfile)
        }
    }
}

parse_options <- function() {

    option_list <- list(
        make_option(c("--f"), action = "store", dest = "indir",
                    default = NULL, help = "Input directory"),
        make_option(c("--o"), action = "store", dest = "outdir",
                    default = NULL, help = "Output directory")
        )

    opt <- parse_args(OptionParser(option_list=option_list))
    return(opt)
}

load_packages <- function() {
    suppressPackageStartupMessages(library("stringr"))
    suppressPackageStartupMessages(library("optparse"))
    return
}


main()
